networkx/generators/small.py

```diff
@@ -768,7 +771,55 @@ def petersen_graph(create_using=None):
     G.name = "Petersen Graph"
     return G
 
+@nx._dispatchable(graphs=None, returns_graph=True)
+def generalized_petersen_graph(n, k, create_using=None):
+    """
+    Returns the generalized Petersen graph GP(n,k).
+
+    The Generalized Peterson graph consists of an outer cycle of n nodes
+    connected to an inner circulant graph of n nodes, where nodes in the
+    inner circulant are connected to their kth nearest neighbor [1]_. A
+    generalized Petersen graph is cubic with 2n nodes and 3n edges. 
+    
+    Some well known graphs are examples of generalized Petersen graphs such
+    as the Desargues graph GP(10,3) and dodecahedron graph GP(10,2).
+
+    Parameters
+    ----------
+    n : int
+       Number of nodes in the outer cycle and inner circulant. n must be greater or equal to 3.
```

Comment:
Also, can you say briefly somewhere what a circulant is?
```suggestion
       Number of nodes in the outer cycle and inner circulant. ``n >= 3`` is required.
```